src/commonMain/kotlin/fr/acinq/lightning/blockchain/electrum/ElectrumMiniWallet.kt

```diff
@@ -0,0 +1,119 @@
+package fr.acinq.lightning.blockchain.electrum
+
+import fr.acinq.bitcoin.*
+import fr.acinq.lightning.utils.Connection
+import fr.acinq.lightning.utils.lightningLogger
+import fr.acinq.lightning.utils.sat
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+import kotlin.native.concurrent.ThreadLocal
+
+data class WalletState(val addresses: Map<String, List<UnspentItem>>) {
```

Comment:
The hierarchy is useful to to update the utxos (Electrum will return a set of utxos per script).